pig/src/main/java/org/jboss/pnc/bacon/pig/impl/repo/RepoManager.java

```diff
@@ -388,8 +401,66 @@ private void addExtraFiles(File m2Repo) {
         }
     }
 
+    /**
+     * Generates the offline manifest text file, which contains all the dependencies, including third party, non-RH
+     * dependencies. First, it adds the build artifacts for all the builds. Then, it executes a query to PNC, requesting
+     * the third party dependencies for each build. Finally, the method generates a file, containing version paths and
+     * checksums, where we can get it, for all build and dependency attributes. The offline manifest file in combination
+     * with the offliner tool (https://repo1.maven.org/maven2/com/redhat/red/offliner/offliner) are used to download all
+     * the attributes to a local maven repository.
+     */
+
     @Override
     public void close() {
         buildInfoCollector.close();
     }
+
+    private void generateOfflinerManifest() {
+        log.info("Will generate the Offline manifest");
+        RepositoryData result = new RepositoryData();
+        List<ArtifactWrapper> artifactsToListRaw = new ArrayList<>();
+        for (PncBuild build : builds.values()) {
+            artifactsToListRaw.addAll(build.getBuiltArtifacts());
+            // TODO: Add filter, basing on the targetRepository.repositoryType, when
+            // https://projects.engineering.redhat.com/browse/NCL-6079 is done
```

Comment:
Don't put internal references in external code. Enter a new ticket internally to track future changes once 6079 is done please.